l10n_it_intrastat_statement/wizard/export_file.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import base64
+
+from odoo import _, fields, models
+
+
+class AccountIntrastatExportFile(models.TransientModel):
+    _name = "account.intrastat.export.file"
+    _description = "Intrastat export file"
+
+    name = fields.Char(string="File Name", readonly=True)
+    data = fields.Binary(string="File", readonly=True)
+    state = fields.Selection(
+        selection=[("choose", "Choose"), ("get", "Get")],
+        default="choose",
+    )
+
+    def act_getfile(self):
+        self.ensure_one()
+        statement_id = self.env.context.get("active_id")
+        statement = self.env["account.intrastat.statement"].browse(statement_id)
+
+        file = statement.generate_file_export()
+
+        filename = statement._get_file_name()
+
+        out = base64.encodebytes(file.encode())
+
+        if self.env.ref(
+            "l10n_it_intrastat_statement.wizard_account_intrastat_export_file"
+        ):
+            view_id = self.env.ref(
+                "l10n_it_intrastat_statement.wizard_account_intrastat_export_file"
+            ).id
+        else:
+            view_id = False
```

Comment:
È possibile evitare di cercare più volte lo stesso riferimento esterno? Va bene che sono nella cache ma è comunque un'operazione che si può evitare.
Penso che questo sia più vicino al codice del modulo prima della migrazione, cosa ne pensi?
```suggestion
        view = self.env.ref(
            "l10n_it_intrastat_statement.wizard_account_intrastat_export_file"
        )
        view_id = view.id
```

Reply:
fatto